eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -93,10 +93,22 @@ class MultiPartHandler(nodeParams: NodeParams, register: ActorRef, db: IncomingP
                   pendingPayments = pendingPayments + (p.add.paymentHash -> (record.paymentPreimage, handler))
               }
           }
-          case None =>
-            Metrics.PaymentFailed.withTag(Tags.Direction, Tags.Directions.Received).withTag(Tags.Failure, "InvoiceNotFound").increment()
-            val cmdFail = CMD_FAIL_HTLC(p.add.id, Right(IncorrectOrUnknownPaymentDetails(p.payload.totalAmount, nodeParams.currentBlockHeight)), commit = true)
-            PendingRelayDb.safeSend(register, nodeParams.db.pendingRelay, p.add.channelId, cmdFail)
+          case None => p.payload.keySend match {
+            case Some(keySend) =>
+              val paymentPreimage = keySend.paymentPreimage
+              val amount = Some(p.payload.totalAmount)
+              val paymentHash = Crypto.sha256(paymentPreimage)
+              val desc = "Donation"
+              // Insert a fake invoice and then restart the incoming payment handler
+              val paymentRequest = PaymentRequest(nodeParams.chainHash, amount, paymentHash, nodeParams.privateKey, desc)
```

Comment:
I think you need to set the invoice features like we do in `ReceivePayment`.
Right now you're not setting support for multipart payments even if it's activated in `nodeParams.features`, which means you will reject a payment if it's sent in multiple HTLCs (which I think lnd may do if they're unable to send the whole amount in a single HTLC).

You'll want something like:

```
val features =  if (nodeParams.features.hasFeature(Features.BasicMultiPartPayment)) {
    Seq(Features.BasicMultiPartPayment.optional, Features.PaymentSecret.optional, Features.VariableLengthOnion.optional)
} else {
    Seq(Features.PaymentSecret.optional, Features.VariableLengthOnion.optional)
}
val paymentRequest = PaymentRequest(nodeParams.chainHash, amount, paymentHash, nodeParams.privateKey, desc, features = features)
```

Reply:
Done